listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -239,6 +240,69 @@ def user_feed(user_name: str):
     }})
 
 
+@user_timeline_event_api_bp.route("/user/<user_name>/feed/events/delete", methods=['OPTIONS', 'POST'])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def delete_feed_events(user_name):
+    '''
+    Delete those events from user's feed that they created themselves.
+    Supports deletion of recommendation, notification and pin events.
+    Along with the authorization token, post one of the following, according
+    to your need.
+    {
+        "event_type": "recording_recommendation",
+        "id": int
+    }
+    {
+        "event_type": "notification",
+        "id": int
+    }
+    {
+        "event_type": "recording_pin",
+        "row_id": int
+    }
+    :param user_name: The MusicBrainz ID of the user from whose timeline events are being deleted
+    :type user_name: ``str``
+    :statuscode 200: Successful deletion
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    '''
+    # TODO: maybe implement logging once approved
+    user = validate_auth_header()
+    if user_name != user['musicbrainz_id']:
+        raise APIUnauthorized("You don't have permissions to delete from this user's timeline.")
+
+    try:
+        event = ujson.loads(request.get_data())
+
+        if event["event_type"] in [UserTimelineEventType.RECORDING_RECOMMENDATION.value,
+                UserTimelineEventType.NOTIFICATION.value]:
+            try:
+                event_deleted = db_user_timeline_event.delete_user_recommendation_notification_event(event["id"], user["id"])
+            except Exception as e:
+                raise APIInternalServerError("Something went wrong. Please try again")
+            if not event_deleted:
+                raise APINotFound("Cannot find '%s' event with id '%s' for user '%s'" % (event["event_type"], event["id"],
+                    user["id"]))
+            return jsonify({"status": "ok"})
+
+        elif event["event_type"] == UserTimelineEventType.RECORDING_PIN.value:
```

Comment:
This should probably be removed as Pinned Recordings have their own endpoint to support deletion.